splunk_eventgen/lib/plugins/generator/jinja.py

```diff
@@ -186,20 +191,34 @@ def gen(self, count, earliest, latest, samplename=None):
                 if self._sample.jinja_count_type in ["line_count", "cycles", "perDayVolume"]:
                     self.jinja_count_type = self._sample.jinja_count_type
             startTime = datetime.datetime.now()
-            working_dir, working_config_file = os.path.split(self.config.configfile)
+
+            # if eventgen is running as Splunk app the configfile is None
+            if self.config.configfile:
+                working_dir, working_config_file = os.path.split(self.config.configfile)
+            else:
+                splunk_home = os.environ["SPLUNK_HOME"]
+                app_name = getattr(self._sample, 'app', 'SA-Eventgen')
+                working_dir = os.path.join(splunk_home, 'etc', 'apps', app_name, 'default')
+
             if not hasattr(self._sample, "jinja_template_dir"):
-                template_dir = "templates"
+                template_dir = os.path.join(os.path.dirname(working_dir), 'samples')
```

Comment:
I specifically did not want to put this in the samples dir, samples should contain the raw events used for parsing, where this is a new type of file.  I'm alright if we use "samples/templates", but please don't put it in the base of samples.

Reply:
Make sense. Update the default jinja template folder to `<bundle>/samples/templates`.